tests/ert_tests/ert3/stats/test_distributions.py

```diff
@@ -237,3 +278,67 @@ def test_uniform_ppf(lower, upper, q, size, index):
     assert sorted(dist.index) == sorted(ppf_result.index)
     for idx in dist.index:
         assert ppf_result.data[idx] == pytest.approx(expected_value)
+
+
+@pytest.mark.parametrize(
+    ("values", "q", "expected_val"),
+    (
+        ([0], 0.0, np.nan),
+        ([0], 0.1, 0),
+        ([0], 0.2, 0),
+        ([0], 0.5, 0),
+        ([0], 0.9, 0),
+        ([0], 1.0, 0),
+        ([1], 0.9, 1),
+        ([0, 0], 0.1, 0),
+        ([0, 0], 0.9, 0),
+        ([42], 0.999, 42),
+        ([0], 1.1, np.nan),
+        ([0], 0.1, 0),
+        ([42, 420, 4200], 0.0, np.nan),
+        ([42, 420, 4200], 0.2, 42),
+        ([42, 420, 4200], 0.3333333, 42),
+        ([42, 420, 4200], 0.3333334, 420),
+        ([42, 420, 4200], 0.5, 420),
+        ([42, 420, 4200], 0.6666666, 420),
+        ([42, 420, 4200], 0.6666667, 4200),
+        ([42, 42, 4200], 0.6666666, 42),
+        ([42, 42, 4200], 0.6666667, 4200),
+        ([42, 4200, 42], 0.6666666, 42),
+        ([42, 4200, 42], 0.6666667, 4200),
+        ([42, 420, 4200], 0.9, 4200),
+        ([42, 420, 4200], 0.9, 4200),
+        ([4200, 420, 42], 0.9, 4200),
+        ([42, 420, 4200], 1.1, np.nan),
+    ),
+)
+def test_discrete_ppf(values, q, expected_val):
+    # Test distributions with a size argument, yielding implicit indices:
+    for size in (1, 5, 10):
+        dist = ert3.stats.Discrete(values, size=size)
+        assert len(dist.index) == size
+
+        ppf_result = dist.ppf(q)
+        assert ppf_result.index == dist.index
+        assert np.array_equal(ppf_result.data, [expected_val] * size, equal_nan=True)
+
+    # Test distribution with explicit index values:
+    for index in (
+        ("a", "b", "c"),
+        ("b", "a", "c"),
+        ("single_key",),
+        tuple(index_len * "x" for index_len in range(1, 10)),
+    ):
+        dist = ert3.stats.Discrete(values, index=index)
+
+        expected = [expected_val] * len(index)
+        assert len(dist.index) == len(expected)
+
+        ppf_result = dist.ppf(q)
+        assert ppf_result.index == dist.index
```

Comment:
re-add sorting here, we don't want to rely on dictionaries to be sorted.

Reply:
no, I think we might as well require sorting here. The indices are treated as tuples and we never really create this as a dictionary where relaxed sorting may apply. @xjules ?

Reply:
In the first case, we don't require `sorted` since indices are not provided and thus the else branch applies in here:
```python
def _build_record_index(
    data: numerical_record_data,
) -> Tuple[Any, ...]:
    if isinstance(data, MutableMapping):
        return tuple(data.keys())
    else:
        return tuple(range(len(data)))
```
but when in comes to providing dict I guess we can't guarantee the same order.

Reply:
As pr. off-github comment, indexed results undergo a `dict` construction in `_to_record()` so we cannot guarantee order. Reinstating the `sorted` in the test.